services/src/main/java/org/keycloak/broker/oidc/mappers/UsernameTemplateMapper.java

```diff
@@ -83,10 +90,20 @@
         property = new ProviderConfigProperty();
         property.setName(TEMPLATE);
         property.setLabel("Template");
-        property.setHelpText("Template to use to format the username to import.  Substitutions are enclosed in ${}.  For example: '${ALIAS}.${CLAIM.sub}'.  ALIAS is the provider alias.  CLAIM.<NAME> references an ID or Access token claim.");
+        property.setHelpText("Template to use to format the username to import.  Substitutions are enclosed in ${}.  For example: '${ALIAS}.${CLAIM.sub}'.  ALIAS is the provider alias.  CLAIM.<NAME> references an ID or Access token claim. \n"
+          + "The substitution can be converted to upper or lower case by appending |uppercase or |lowercase to the substituted value, e.g. '${CLAIM.sub | lowercase}");
         property.setType(ProviderConfigProperty.STRING_TYPE);
         property.setDefaultValue("${ALIAS}.${CLAIM.preferred_username}");
         configProperties.add(property);
+
+        property = new ProviderConfigProperty();
+        property.setName(TARGET);
+        property.setLabel("Target");
+        property.setHelpText("Destination username to apply the mapper to. LOCAL means that the changes are applied to the username stored in local database upon import. BROKER_ID and BROKER_USERNAME means that the changes are stored into the ID or username used for federation lookup, respectively.");
```

Comment:
From the tooltip, it is not so obvious to me what Target is used for. However I don't have better idea how to improve this as it is not trivial to describe... So this is just a feedback for the case you can come with bit clearer description, but feel free to ignore this comment if not :)

Same comment applies for SAML mapper too.